### DISCUSSION TITLE
Tailwind CLI --watch mode is broken for html files.

### DISCUSSION THREAD

**Discussion by doyouknowaboutq** [tailwindlabs/tailwindcss#6885] [Help] [answered]
**What version of Tailwind CSS are you using?**
v3.0.9

**What version of Node.js are you using?**
v17.3.0

**What version of NPM are you using?**
v8.3.0

**What operating system are you using?**
ArchLinux

**Describe your issue**
Tailwindcss doesn't rebuild output.css properly while watching changes on index.html. Let's say i have a div with **_bg-red-200_** class. If i change it to **_bg-red-300_** and save the index.html, the rebuilding process is working but not the properly way. Because it is appending **_bg-red-300_** class to end of the output css file while **_bg-red-200_** class standing there. When i change it one more time with for example **_bg-red-100_** nothing happens this time. Rebuilding process working correctly just only when i save the source.css or tailwind.config.js files.

**Comment by adamwathan** [comment, marked as answer]
Hey! In watch mode, Tailwind doesn't rebuild from scratch on each save, it only adds things you haven't already used like you're noticing. This is for performance reasons so we don't have to rescan all of your template files again. When you save your CSS file or config file, we do a total rebuild out of necessity (because the classes that Tailwind is looking for might be different with a different config) which is a bit slower, but that's why you see the unused classes disappear then.

Not sure what could be causing the `bg-red-100` issue with it just not showing up, but if you provide a reproduction happy to look.

**Reply by doyouknowaboutq** [reply]
Actually I thought it might have been done on purpose for performance reasons. But as you said the the bg-red-100 issue confused me. Because I'm pretty sure it wasn't added or even later changes. After deleting and reinstalling everything, including npm itself, I was happy to see that the problem was completely fixed on a project that was built from scratch. There was no change in versions. Despite this, the problem still persists in my active project. Therefore, this situation only affects me for reasons arising again from me and I have to find the solution myself. Apologies for the false alarm and a huge thank you to everyone who contributed to this amazing project.
